### PR TITLE
Cuda versions

### PR DESCRIPTION
**Pull request by liruipeng**
This PR adds support and regression tests for all the versions from CUDA 9.0 to 12.0.

- [x] tux
- [x] lassen
- [x] tioga
- [x] sunspot

A new test `machine-lassen-cuda.sh` is added, which takes about 1 hour on lassen. We should run it in the regression to make sure all versions of CUDA are OK.

### CODE REVIEW

**Comment by liruipeng** [comment]
Should go after #871. Maybe also better to review after #871 being merged.

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

**Review by victorapm — APPROVED** [review]
Thanks, Rui Peng!

File: src/test/TEST_error/error.saved
Comment (victorapm) [inline src/test/TEST_error/error.saved:1]: @liruipeng I also need these changes in the `ilu-amd` branch. Are you close to merge `cuda_versions` into master? If that's the case, I would wait for your merge

Comment (liruipeng) [reply src/test/TEST_error/error.saved:1]: will do it tonight . Thanks!

**Pull request merged**
